ideas/292-decouple-whisper-key-from-wallet-key.md

```diff
@@ -0,0 +1,177 @@
+> Here all roles in swarm are defined and filled, one of the contributors should
+> responsibility of the Idea as Lead.
+>
+> Lead Contributor is the Owner of the Idea. If required, they can get support
+> from a PM, but should be responsible for end to end execution of the Idea.
+> This includes ensuring appropriate resources are allocated, setting realistic
+> timelines and milestones, and any post-launch metrics or bug fixes that are
+> attributed to the Idea
+>
+> A swarm requires at minimum 3 contributors. For user-facing swarms
+> PM/UX(R)/Eng functions have to be present.
+>
+> For swarms lead by core contributors, swarm lead is evaluator by default.
+>
+> 'Contributor' should be replaced with a descriptive role type.
+
+- Lead Contributor: [@pilu](https://github.com/pilu)
+- Contributor: [@cammellos](https://github.com/cammellos)
+- Evaluator (defaults to lead contributor):
+- Contributor:
+- Contributor:
+- QA:
+- PM (required for user-facing):
+- UX(R) (required for user-facing swarms):
+
+## Product Overview
+
+In the current implementation, the same key pair is used for the whisper identity and the wallet.
+The Status contact code is an uncompressed public key, and users share it with other people to be contacted.
+In public chats, anyone can see the public key of a user by clicking on  the avatar near the message.
+
+### Problem
+
+Since the whisper key pair is the same that we use for the wallet, it means that from the Status contact code, anyone can derive the wallet address and easily check a user's balance and transaction history.
+Any user with a fair amount of value in their wallet can be target of phishing attacks, especially in public chats where any malicious user can track the wallets of all the active users.
+
+### Technical solution
+
+Status already implements a Hierarchical Deterministic wallet (HD Wallet) that allows us to derive multiple determinist keys from the same 12 mnemonic words.
+For the wallet keys, we can continue to use the current derivation path, which is the one described by BIP44: `m/44'/60'/0'/0/0`.
+We can then extend the key store structure to have an additional key used for the whisper identity and derived with a different derivation path that needs to be decided.
+
+### Changes in the chat/wallet flow
+
+In the current implementation, users can send ETH/SNT via the chat without asking the wallet address of the recipient, since it can be easily derived from the chat contact code, which is the public key.
+If we add a different key for the chat/whisper identity, we need to implement a protocol such that user A need to request the address of user B before being able to start a transaction. User B can decide to approve the request and reveal or not the wallet address.
+
+### Compatibility
+
+The key pair used for the wallet will remain the same, so the wallet address won't change.
+
+The chat contact will change so after the upgrade, users will need to exchange their contact again to be able to chat.
```

Comment:
Will there be any UX solution to let the user know that there 1:1 chat will not work anymore? Maybe we can mark all existing 1:1 chats as obsolete and show some message in those ones.